packages/keyring-api/src/api/v2/wrapper/keyring-wrapper.ts

```diff
@@ -0,0 +1,192 @@
+import type { Keyring, AccountId } from '@metamask/keyring-utils';
+import type { Json } from '@metamask/utils';
+
+import { KeyringAccountRegistry } from './keyring-account-registry';
+import type {
+  CreateAccountOptions,
+  ExportAccountOptions,
+  ExportedAccount,
+} from '..';
+import type { KeyringAccount } from '../../account';
+import type { KeyringRequest } from '../../request';
+import type { KeyringV2 } from '../keyring';
+import type { KeyringCapabilities } from '../keyring-capabilities';
+import type { KeyringType } from '../keyring-type';
+
+/**
+ * Basic options for constructing a {@link KeyringWrapper}.
+ */
+export type KeyringWrapperOptions<InnerKeyring extends Keyring> = {
+  /**
+   * The underlying "old" keyring instance that this wrapper adapts.
+   */
+  inner: InnerKeyring;
+
+  /**
+   * The concrete keyring type exposed through the V2 interface.
+   */
+  type: KeyringType;
+
+  /**
+   * Capabilities of the underlying keyring.
+   */
+  capabilities: KeyringCapabilities;
+
+  /**
+   * Identifier for the entropy source associated with this keyring.
+   */
+  entropySourceId: string;
```

Comment:
Not sure why we have this here? HD keyrings do have an entropy source, but private key keyrings have not, which mean we won't be able to provide one if we re-use the wrapper for those.

This should probably live in the `HdKeyringV2` directly?